src/features/metadata-instance-editor/fields/DateField.js

```diff
@@ -18,29 +19,41 @@ type Props = {
     onRemove: (key: string) => void,
 };
 
-const DateField = ({ dataKey, dataValue, displayName, description, intl, onChange, onRemove }: Props) => (
-    <DatePicker
-        className="metadata-instance-editor-field-date"
-        dateFormat="utcISOString"
-        description={description}
-        displayFormat={{
-            day: 'numeric',
-            month: 'long',
-            year: 'numeric',
-        }}
-        hideOptionalLabel
-        label={displayName}
-        onChange={(date: Date, isoDate: string) => {
-            if (isoDate) {
-                onChange(dataKey, isoDate);
-            } else {
-                onRemove(dataKey);
-            }
-        }}
-        placeholder={intl.formatMessage(messages.metadataFieldSetDate)}
-        value={typeof dataValue === 'string' ? convertISOStringToUTCDate(dataValue) : undefined}
-    />
-);
+const DateField = ({ dataKey, dataValue, displayName, description, intl, onChange, onRemove }: Props) => {
+    // if the JS engine thinks there's an invalid date, convert to known format if possible
+    let parsedDataValue = dataValue;
+
+    try {
+        parsedDataValue =
+            !isNaN(dataValue) && typeof dataValue === 'string' ? convertISOStringtoRFC3339String(dataValue) : dataValue;
```

Comment:
Can this logic be encapsulated in the convert function? Otherwise I would assume every client/component will need to duplicate it.

Reply:
Yah, I considered that. the conversion is harmless, but I also didn't want to apply it globally on LN53. If I embed this check in the conversion function, I'd have to use it for every DatePicker, even for browsers that support the ISO8601 date formats.

It occurred to me a better way to use this might be to embed it in the existing conversion functions instead of importing it in the client components. I'll try that for comparison